### PR TITLE
Update Keras categorical input design, mark it as complete.

### PR DESCRIPTION
**Pull request by tanzhenyu**
Only one minor change to split "IndexLookup" to "StringLookup" and "IntegerLookup". The rest is trivial.

But more importantly, add an official guide on how TF1 Estimator users with feature columns can convert their code into TF2 Keras users with KPL.

### CODE REVIEW

File: rfcs/20191212-keras-categorical-inputs.md [hunk 1]
```diff
@@ -19,237 +19,407 @@ The proposed layers should support ragged tensors.
 
 ## Motivation
 
-Specifically, by introducing the 4 layers, we aim to address these pain points:
+Specifically, by introducing the 5 layers, we aim to address these pain points:
 * Users have to define both feature columns and Keras Inputs for the model, resulting in code duplication and deviation from DRY (Do not repeat yourself) principle. See this [Github issue](https://github.com/tensorflow/tensorflow/issues/27416).
 * Users with large dimension categorical inputs will incur large memory footprint and computation cost, if wrapped with indicator column through `tf.keras.layers.DenseFeatures`.
-* Currently there is no way to correctly feed Keras linear model or dense layer with multivalent categorical inputs or weighted categorical inputs.
+* Currently there is no way to correctly feed Keras linear model or dense layer with multivalent categorical inputs or weighted categorical inputs, or shared embedding inputs.
+* feature columns offer black-box implementations, mix feature engineering with trainable objects, and lead to
```
Comment (goldiegadde) [inline rfcs/20191212-keras-categorical-inputs.md:26]: nit: feature/Feature

File: rfcs/20191212-keras-categorical-inputs.md [hunk 2]
```diff
@@ -19,237 +19,407 @@ The proposed layers should support ragged tensors.
...
 ```
 
-### Workflow 2
+Note the hashed output from KPL will be different than the hashed output from feature column, given how seed is choosen. `Hashing` also supports customized `salt`.
 
-The second example gives an instruction on how to transition from categorical feature columns to the proposed layers. Note that one difference for vocab categorical column is that, instead of providing a pair of mutually exclusive `default_value` and `num_oov_buckets` where `default_value` represents the value to map input to given out-of-vocab value, and `num_oov_buckets` represents value range of [len(vocab), len(vocab)+num_oov_buckets) to map input to from a hashing function given out-of-vocab value. In practice, we believe out-of-vocab values should be mapped to the head, i.e., [0, num_oov_tokens), and in-vocab values should be mapped to [num_oov_tokens, num_oov_tokens+len(vocab)).
+2. `tf.feature_column.categorical_column_with_identity`
+This feature column is merely for having identical inputs and outputs except mapping out-of-range value into `default_value`, thus can easily be done at data cleaning stage,
+not be part of feature engineering, and hence dropped in this proposal.
```
Comment (goldiegadde) [inline rfcs/20191212-keras-categorical-inputs.md:56]: nit: for better readability, do you want to move this to the end , since this is not replaced and deemed not needed in feature engineering.

File: rfcs/20191212-keras-categorical-inputs.md [hunk 3]
```diff
@@ -1,15 +1,15 @@
 # Keras categorical inputs
 
-| Status        | Accepted                                             |
+| Status        | Completed                                            |
```
Comment (ematejska) [inline rfcs/20191212-keras-categorical-inputs.md:3]: Do you mean it's implemented?  I would prefer the terminology Implemented instead of Completed.  Could you also post the PR number where it was implemented?  Something like:

Status: Implemented (Link to PR)

File: rfcs/20191212-keras-categorical-inputs.md (hunk 2 above)
Comment (tanzhenyu) [reply rfcs/20191212-keras-categorical-inputs.md:56]: Done.

File: rfcs/20191212-keras-categorical-inputs.md (hunk 3 above)
Comment (tanzhenyu) [reply rfcs/20191212-keras-categorical-inputs.md]: Done.

File: rfcs/20191212-keras-categorical-inputs.md (hunk 1 above)
Comment (tanzhenyu) [reply rfcs/20191212-keras-categorical-inputs.md]: Done.

File: rfcs/20191212-keras-categorical-inputs.md (hunk 2 above)
Comment (goldiegadde) [reply rfcs/20191212-keras-categorical-inputs.md:56]: sg!
